src/imgui/imgui_raii_wrapper.h

```diff
@@ -0,0 +1,162 @@
+#ifndef MYPROJECT_IMGUI_RAII_WRAPPER_H
+#define MYPROJECT_IMGUI_RAII_WRAPPER_H
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <imgui.h>
+
+namespace irw {
+
+class Windowzita {
+public:
+  [[nodiscard]] explicit Windowzita(std::string const& id) : selected{ImGui::Begin(id.c_str())} {}
+  Windowzita(Windowzita const&) = delete;
+  Windowzita& operator=(Windowzita const&) = delete;
+  Windowzita(Windowzita&&) = delete;
+  Windowzita& operator=(Windowzita&&) = delete;
+  ~Windowzita() {
+    ImGui::End();
+  }
```

Comment:
Well, what is this? Really, what is? :D
I will start to make really noob questions in order to be a part of the project. To try to make the project more cpp-noob-friendly.

Reply:
Good question. The first is a constructor and the other five together with the no-argument constructor are called the special member functions. Search for this on google and you will find many resources explaining what each one does, but this may be a bit advanced. If I were you, I would focus on the first one (constructor) and the last one (destructor). Notice all the other ones have "= delete", meaning we're not even using them. The reason I wrote them is because I was following the rule of 0/5. You can search for this but I think you should leave it for the future. First, make sure you understand constructor and destructor. Constructors are in almost every language, so learn it well.